app/scripts/services/secrets.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("SecretsService", function(){
+
+    var groupSecretsByType = function(secrets, addEmptyValue) {
```

Comment:
Was thinking about it as well, but then I've realized that we are using the function od several places and wanted to spent some lines by putting the logic into the service itself.. 
Will update :) thanks :+1: